### PR TITLE
add progress bars

### LINKED ISSUE #4070: [ENH]: Add a progress bar to postprocessing in `_conclude()` functions in MSD.py
Quite a lot of work is done in the `_conclude()` functions (`_conclude_simple()` and `_conclude_fft()`) in `msd.py`. These should have progress bars. 

@SophiaRuan will get first shot at this as I had to close her PR due to some poor planning on my end. (Replacement for #4028)

### PR DESCRIPTION
**Pull request by SophiaRuan**
Fixes #4070 

Changes made in this Pull Request:
 - add progress bars to _conclude()

PR Checklist
------------
 - [x] Tests?
 - [ ] Docs?
 - [x] CHANGELOG updated?
 - [x] Issue raised/referenced?

### CODE REVIEW

**Comment by SophiaRuan** [comment]
Hi @orbeckst @hmacdope, I tried to add progress bars to `_conclude_simple` and `_conclude_fft`. I installed the module progressbar locally and the code pass the `msd_test` locally. How can I install the progressbar for remote repo here or is there any other solution?

**Comment by hmacdope** [comment]
@SophiaRuan normally we use the tqdm package for progress bars rather than "progress bars" You will need to test on your local machines

**Comment by orbeckst** [comment]
@hmacdope sorry to butt in... we do indeed have our own `ProgressBar`! The question is if the customizations in ProgressBar makes sense, it's a thin wrapper around tqdm.

**Comment by SophiaRuan** [comment]
Hi @hmacdope @orbeckst, I am wondering how to pass the dark_lint test. I am not sure if the current code aligns with the requirement.

**Comment by RMeli** [comment]
@SophiaRuan the `dark_lint` test is optional, see #4067. Here it seems that it is only asking to add a blank line after the `from tqdm import tqdm`.

**Comment by hmacdope** [comment]
@RMeli is correct. @SophiaRuan would you be able to share a screenshot showing that the progress bars work? Otherwise looks good to me!

File: package/MDAnalysis/analysis/msd.py [hunk 1]
```diff
@@ -387,7 +387,10 @@ def _conclude_simple(self):
         """
         lagtimes = np.arange(1, self.n_frames)
         positions = self._position_array.astype(np.float64)
-        for lag in lagtimes:
+
+        # progressbar
+        from tqdm import tqdm
```
Comment (IAlibay) [inline package/MDAnalysis/analysis/msd.py]: If all paths lead to an import of tqdm, could you move this up to the top of the file please?

**Review by IAlibay — CHANGES_REQUESTED** [review]

**Comment by SophiaRuan** [comment]
> @RMeli is correct. @SophiaRuan would you be able to share a screenshot showing that the progress bars work? Otherwise looks good to me!

Hi @hmacdope, I am wondering how to check if progress bars work. I only used the `pytest` on my local machine to check if the code pass the test, but I can't see the progress bars.

File: package/MDAnalysis/analysis/msd.py (hunk 1 above)
Comment (SophiaRuan) [reply package/MDAnalysis/analysis/msd.py]: > If all paths lead to an import of tqdm, could you move this up to the top of the file please?

Yeah, I'll make this change. Thank you!

**Comment by hmacdope** [comment]
> > @RMeli is correct. @SophiaRuan would you be able to share a screenshot showing that the progress bars work? Otherwise looks good to me!
> 
> 
> 
> Hi @hmacdope, I am wondering how to check if progress bars work. I only used the `pytest` on my local machine to check if the code pass the test, but I can't see the progress bars.

You should set up an example piece of code (copying one of the tests is fine) and then run it to see if progress bars appear.

**Comment by hmacdope** [comment]
You may need to use the verbose=True flag to run I can't remember.

**Comment by SophiaRuan** [comment]
> 
Hi @hmacdope, I tried to run the `test_msd.py` in Pycharm. Does this look good to you?
Screen Shot 2023-03-16 at 12 03 19

**Review by hmacdope — CHANGES_REQUESTED** [review]
Could you add a CHANGELOG entry?

**Comment by SophiaRuan** [comment]
> #4070
Hi @hmacdope, I just changed the CHANGELOG. Could you have a look? Thank you!
[(https://github.com/MDAnalysis/mdanalysis/compare/develop...SophiaRuan:mdanalysis:patch-1)]

**Comment by hmacdope** [comment]
@SophiaRuan you need to add them in this same PR

File: package/CHANGELOG [hunk 2]
```diff
@@ -13,7 +13,15 @@ The rules for this file:
   * release numbers follow "Semantic Versioning" http://semver.org
 
 ------------------------------------------------------------------------------
-  
+
+03/20/23 SophiaRuan
```
Comment (hmacdope) [inline package/CHANGELOG]: This needs to go with the other 2.5.0 entries, just add it to the list.

**Review by hmacdope — CHANGES_REQUESTED** [review]

File: package/CHANGELOG (hunk 2 above)
Comment (SophiaRuan) [reply package/CHANGELOG]: Hi @hmacdope, I just added the change to the 2.5.0 entries.

**Review by hmacdope — CHANGES_REQUESTED** [review]
Would you also be able to addd your name to `AUTHORS` if you havn't already. other than that looks good!

File: package/CHANGELOG [hunk 3]
```diff
@@ -50,6 +50,9 @@ Changes
     analysis.nucleicacids' NucPairDist and WatsonCrickDist classes has been
     removed. Please use the `results.pair_distances` and `times` attributes
     instead (Issue #3744)
+  * Add progress bars to track the progress of _conclude() functions
+    (_conclude_simple() and _conclude_fft()) in msd.py (Issue #4070, PR #4072)
```
Comment (IAlibay) [inline package/CHANGELOG]: Entries should be newest first (please see line 7 in the instructions at the top of the file). This helps us easily scan the ordered history of changes in the repository.

File: package/CHANGELOG [hunk 4]
```diff
@@ -50,6 +50,9 @@ Changes
...
+
```
Comment (IAlibay) [inline package/CHANGELOG]:
```suggestion
```
Sorry this might look overly pedantic but we try to keep the empty line formatting fixed in this file.

**Review by IAlibay — COMMENTED** [review]
Just a couple of small review comments on changelog formatting from me please.

File: package/CHANGELOG [hunk 5]
```diff
@@ -13,10 +13,11 @@ The rules for this file:
   * release numbers follow "Semantic Versioning" http://semver.org
 
 ------------------------------------------------------------------------------
-  
+
 ??/??/?? IAlibay, pgbarletta, mglagolev, hmacdope, manuel.nuno.melo, chrispfae, 
          ooprathamm, MeetB7, BFedder, v-parmar, MoSchaeffler, jbarnoud, jandom,
-         xhgchen, jaclark5, DrDomenicoMarson
+         xhgchen, jaclark5, DrDomenicoMarson, SophiaRuan
+
```
Comment (hmacdope) [inline package/CHANGELOG]: I know this is annoying @SophiaRuan but this should show no diff **except your name**

**Review by hmacdope — CHANGES_REQUESTED** [review]

File: package/CHANGELOG (hunk 5 above)
Comment (SophiaRuan) [reply package/CHANGELOG]: Hi @hmacdope, I just made some changes. Could you help me check a bit? Thank you for your help!

**Comment by SophiaRuan** [comment]
Hi @hmacdope, I made some changes to the Author and CHANGELOG recently. Could you help me check on this? 
Also, I find there are some conflicts in Author and CHANGELOG constantly. Should I solve the conflicts remotely or on my local machine? Previously, I find I need to solve conflicts both remotely and locally in order to push my next change. Sorry for asking so many questions. Thank you for your help!

File: package/CHANGELOG [hunk 6]
```diff
@@ -13,10 +13,10 @@ The rules for this file:
   * release numbers follow "Semantic Versioning" http://semver.org
 
 ------------------------------------------------------------------------------
-  
+
```
Comment (hmacdope) [inline package/CHANGELOG]:
```suggestion
  
```

**Review by hmacdope — CHANGES_REQUESTED** [review]
Apply this suggestion and should be good to go.

**Comment by SophiaRuan** [comment]
> Apply this suggestion and should be good to go.

Hi @hmacdope, I have made the change! Thank you!

**Review by hmacdope — APPROVED** [review]
Looks good to me !

**Comment by hmacdope** [comment]
@SophiaRuan resolve conflicts and then good to go.

**Comment by SophiaRuan** [comment]
> @SophiaRuan resolve conflicts and then good to go.

Hi @hmacdope @IAlibay, I have solved the conflicts. Thank you for your help!

**Review by IAlibay — APPROVED** [review]

**Pull request merged**